src/Symfony/Component/Validator/Constraints/RangeValidator.php

```diff
@@ -76,5 +76,15 @@ public function validate($value, Constraint $constraint)
                 ->setCode(Range::TOO_LOW_ERROR)
                 ->addViolation();
         }
+
+        if (null !== ($step = $constraint->step) && !\in_array($value, range($min, $max, $step), true)) {
```

Comment:
Shouldn't the last argument of `in_array()` be false? I feel like this should handle stringy numbers, since this is what forms always get as input. (and the rest of the validator works with them already)
Please add a few test cases with strings as values (numbers and not numbers, same for min/max tests).

Reply:
i'm doing it.